app/share.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { View } from 'react-native';
+import { createAppContainer, createStackNavigator, createSwitchNavigator } from 'react-navigation';
+import { Provider } from 'react-redux';
+import firebase from 'react-native-firebase';
+
+import Navigation from './lib/Navigation';
+import store from './lib/createStore';
+import { appInit } from './actions';
+import ShareListView from './views/ShareListView';
+import ShareView from './views/ShareView';
+import SelectServerView from './views/SelectServerView';
+import AuthLoadingView from './views/AuthLoadingView';
+import WithoutServersView from './views/WithoutServersView';
+import sharedStyles from './views/Styles';
+import { isNotch } from './utils/deviceInfo';
+import { HEADER_BACKGROUND, HEADER_TITLE, HEADER_BACK } from './constants/colors';
+
+const defaultHeader = {
+	headerStyle: {
+		backgroundColor: HEADER_BACKGROUND
+	},
+	headerTitleStyle: {
+		color: HEADER_TITLE
+	},
+	headerBackTitle: null,
+	headerTintColor: HEADER_BACK
+};
+
+const InsideNavigator = createStackNavigator({
+	ShareListView,
+	ShareView,
+	SelectServerView
+}, {
+	initialRouteName: 'ShareListView',
+	defaultNavigationOptions: defaultHeader
+});
+
+const OutsideNavigator = createStackNavigator({
+	WithoutServersView
+}, {
+	initialRouteName: 'WithoutServersView'
+});
+
+const AppContainer = createAppContainer(createSwitchNavigator(
+	{
+		OutsideStack: OutsideNavigator,
+		InsideStack: InsideNavigator,
+		AuthLoading: AuthLoadingView
+	},
+	{
+		initialRouteName: 'AuthLoading'
+	}
+));
+
+const getActiveRouteName = (navigationState) => {
+	if (!navigationState) {
+		return null;
+	}
+	const route = navigationState.routes[navigationState.index];
+	if (route.routes) {
+		return getActiveRouteName(route);
+	}
+	return route.routeName;
+};
+
+const onNavigationStateChange = (prevState, currentState) => {
+	const currentScreen = getActiveRouteName(currentState);
+	const prevScreen = getActiveRouteName(prevState);
+
+	if (prevScreen !== currentScreen) {
+		firebase.analytics().setCurrentScreen(currentScreen);
+	}
+};
+
+class Root extends React.Component {
+	constructor(props) {
+		super(props);
+		store.dispatch(appInit());
+		this.state = {
+			isLandscape: false
+		};
+	}
+
+	handleLayout = (event) => {
+		const { width, height } = event.nativeEvent.layout;
+		this.setState({ isLandscape: width > height });
+	}
+
+	render() {
+		const { isLandscape } = this.state;
+		return (
+			<View
+				style={[{ ...sharedStyles.container }, isLandscape && isNotch ? { ...sharedStyles.notchLandscapeContainer } : {}]}
```

Comment:
Why?

Reply:
Because the ```safe-area-view``` -> ```react-navigation``` is a dependency on 'react-native-safe-area-view', in iOS extensions there was an issue in the safe-area-view but this has been fixed in the newer version.

https://github.com/react-native-community/react-native-safe-area-view/issues/66